source/common/http/conn_manager_impl.cc

```diff
@@ -1640,9 +1640,18 @@ void ConnectionManagerImpl::ActiveStream::encodeHeaders(ActiveStreamEncoderFilte
 void ConnectionManagerImpl::ActiveStream::encodeHeadersInternal(ResponseHeaderMap& headers,
                                                                 bool end_stream) {
   // Base headers.
-  if (!stream_info_.hasResponseFlag(StreamInfo::ResponseFlag::ResponseFromCacheFilter)) {
+
+  // By default, always preserve the upstream date response header if present. If we choose to
+  // overwrite the upstream date unconditionally (a previous behavior), only do so if the response
+  // is not from cache
+  const bool should_preserve_upstream_date =
```

Comment:
I'd suggest using
Runtime::runtimeFeatureEnabled("")

Reply:
Should this guard be under the `envoy.reloadable_features` namespace, then?

Reply:
Yeah, this is something we will remove if no one complains.